### PR TITLE
fix(store): Handle release being deleted prior to group creation

### PR DESCRIPTION
**Pull request by macqueen**
fixes SENTRY-40C

not sure if there's a better way to do this...

### CODE REVIEW

File: src/sentry/event_manager.py [hunk 1]
```diff
@@ -844,11 +844,25 @@ def _save_aggregate(self, event, hashes, release, **kwargs):
...
-                ), True
+            try:
+                with transaction.atomic():
+                    short_id = project.next_short_id()
+                    group, group_is_new = Group.objects.create(
+                        project=project, short_id=short_id, **kwargs
+                    ), True
+            except IntegrityError as exc:
+                # it's possible the release was deleted between
+                # when we queried for the release and
```
Comment (macqueen) [inline src/sentry/event_manager.py]: lol will fix this comment

**Comment by mattrobenolt** [comment]
My thought process was to use a QuerySet to represent `first_release` when it goes to insert into database. Not sure if this is complex given where it's at in code or not, but something effectively like:

```
Group.objects.create(
...
first_release=Release.objects.filter(id=first_release_id).values('id')
)
```

This way it would be handled automatically on the database side.

**Comment by macqueen** [comment]
@mattrobenolt hmm so i'm not sure if a queryset is going to work without using `.first()`, which i think forces execution, right? Doing what you suggested yields `ValueError: Cannot assign "[{'id': 1L}]": "Group.first_release" must be a "Release" instance.` 😞

**Comment by mattrobenolt** [comment]
> hmm so i'm not sure if a queryset is going to work without using .first(), which i think forces execution, right

Yeah, so we might have to do this. I wonder if there's no way to actually use a QuerySet here. That might even be invalid SQL. I'll poke at it since that's ideal, but an extra query here won't really hurt.

It still won't be perfect, but it'll reduce the window for error.

**Comment by macqueen** [comment]
ok, updated to use `first` for now

**Review by mattrobenolt — APPROVED** [review]
Should be fine.

**Pull request merged**